### PR TITLE
Feature/router

### LINKED ISSUE #9: Make form dates dynamic, since Earth API often returns 404 for very recent dates, prevent the user from making GETs within the last 7 months. Allow the Apod date form to use all dates.

### PR DESCRIPTION
**Pull request by BrandonAinsworth**
What does this PR do?

- Continues to keep the Form component modular by conditionally setting the max date allowed on the input form depending on user location.
- Adds loading gif
- Adds ImageMe to NavBar 

Where should the reviewer start?
-Line 4 Form.js

How should this be manually tested?
-This can currently only be tested by using the application in its current state.

What are the relevant tickets?
- Resolves #9

### CODE REVIEW

File: src/Form.js [hunk 1]
```diff
@@ -15,25 +17,38 @@ const configureDate = (date) => {
 const submissionType = (e) => {
   console.log(location.pathname)
   if(location.pathname === '/') {
+    setLoading(true)
```
Comment (jordan-sullivan) [inline src/Form.js:20]: Nice addition of setLoading here!

File: src/App.js [hunk 2]
```diff
@@ -79,14 +84,20 @@ const App = () => {
...
+     <p>Your coordinates are:<br></br>
+      Latitude: {userLat}<br></br>
+      Longitude: {userLong}<br></br>
+    </p>
      <button value= { imageURL } onClick= { saveImage }> Save this Image! 
      </button>
      </> 
-    :
+    :        
+    loading ? <img src={asset}></img> : <>
```
Comment (jordan-sullivan) [inline src/App.js:95]: great error handling with your loading hook.

File: src/Form.js [hunk 3]
```diff
@@ -15,25 +17,38 @@ const configureDate = (date) => {
...
+  } else {
+    setFormDate(dayjs().format('YYYY/MM/DD').split('/').join('-'))
+  }
+}
+
+useEffect(() => {
+  console.log('hello form')
+  console.log(formDate)
+  maxDateMaker()
+},[maxDateMaker, formDate])
```
Comment (jordan-sullivan) [inline src/Form.js:39]: Nice job getting everyting done in this one maxdateMaker function. You may want to look into if you actually need a useEffect hook here- you might be able to do this functionality in just another function.

Comment (jordan-sullivan) [inline src/Form.js:39]: Nice job getting everything done in this one maxDateMaker function. You may want to look into if you actually need a useEffect hook here- you might be able to do this functionality in just another function.

File: src/Form.js (hunk 1 above)
Comment (jordan-sullivan) [inline src/Form.js:20]: Nice addition of setLoading here!

File: src/App.js (hunk 2 above)
Comment (jordan-sullivan) [inline src/App.js:95]: great error handling with your loading hook.

**Review by jordan-sullivan — APPROVED** [review]
Great work so far Brandon! I left a few comments on Form and in App. Love your loading graphic and your error handling with the Loading set up. Nice work on the MaxDates as well! That seemed tricky to navigate! -Jordan

File: src/Form.js (hunk 3 above)
Comment (BrandonAinsworth) [reply src/Form.js:39]: Thank you! My theory around the useEffect hook here was to force a re-render of the Form when the max date changes, so as not to accidentally carry over the wrong max value to the wrong page. I did initially try setting the max to a function value from Form, but this caused a few errors. Fortunately, this useEffect worked on its initial implementation!

**Pull request merged**